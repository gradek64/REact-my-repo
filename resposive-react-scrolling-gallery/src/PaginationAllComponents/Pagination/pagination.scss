$ballSize: 50rem;
$ballColor: green;
$textColor: white;
$textSize: 0.5rem;

.ball {
  color: $textColor;
  font-size: $textSize;
  margin-right: 1rem;
  margin-top: 1rem;
  padding: 5px 8px;
  background-color: $ballColor;
  border-radius: 50%;
  cursor: pointer;
  border: 1px solid transparent;

  &:hover {
    background-image: linear-gradient(green, orange);
    border: 1px solid orange;
  }
}