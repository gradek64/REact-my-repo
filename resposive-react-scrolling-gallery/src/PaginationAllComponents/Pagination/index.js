import React from 'react';
import PropTypes from 'prop-types';
// custom
import './pagination.scss';


const Pagination = ({ howMany, eventHandler }) => (
  <div>
    {
      Array.from({ length: howMany }, (x, i) => (
        <button
          key={`ball${i}`}
          type="button"
          className="ball"
          onClick={() => { eventHandler(i); }}
        >
          {i + 1}
        </button>
      ))
    }
  </div>);

Pagination.propTypes = {
  howMany: PropTypes.number.isRequired,
  eventHandler: PropTypes.func.isRequired,
};

export default Pagination;
