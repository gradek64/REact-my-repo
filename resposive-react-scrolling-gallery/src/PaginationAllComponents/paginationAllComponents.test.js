import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';
// import hook function
import PagginationAllCompHook from './index';


// mock reack hook setState dont exist in Enzyme yet;
const setState = jest.fn();
const useStateSpy = jest.spyOn(React, 'useState');
useStateSpy.mockImplementation(init => [init, setState]);

/*
    hook can not be called just like normal functions (isolated from React)
    instead you need to wrap them into container
    and give prop hook that passes hook function itself;
*/
function HookWrapper(props) {
  const hook = props.hook ? props.hook() : undefined;
  return <div hook={hook} />;
}


test('should render correctly', () => {
  const wrapper = shallow(<HookWrapper hook={() => PagginationAllCompHook()} />);
  expect(wrapper).toMatchSnapshot();
});

it('should set init value and update setState', () => {
  const wrapper = shallow(<HookWrapper hook={() => PagginationAllCompHook()} />);

  // it only uses this next 2 lines below not needed yet it is a mock form Ract setState hook
  const setState = jest.fn((e) => { console.log('w', e); });
  /*  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation(init => [init, setState]);
  */

  // those are props send to the hook wrapper
  const { hook } = wrapper.find('div').props();

  /*
      first you need to render Pagination component since Hook component
      doesnt go that deep event with mount render;
  */
  const PaginatioWrapper = shallow(<Pagination howMany={2} eventHandler={setState} />);
  expect(PaginatioWrapper).toMatchSnapshot();

  // once you have access to pagination buttons you need to simulate click
  PaginatioWrapper.find('button').at(0).simulate('click');

  // then expect to change you hook state for example call setState with param = 0
  expect(setState).toHaveBeenCalledWith(0);

  // the same for the second button render from how many props
  PaginatioWrapper.find('button').at(1).simulate('click');

  // then expect to change you hook state for example call setState with param = 1
  expect(setState).toHaveBeenCalledWith(1);
});
