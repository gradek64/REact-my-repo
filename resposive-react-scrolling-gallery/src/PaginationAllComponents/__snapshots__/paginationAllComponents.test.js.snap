// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  hook={
    <div
      className="paginationListContainer"
    >
      <PaginatedList
        list={
          Array [
            Object {
              "_id": "593e9297e17df20c4a237d42",
              "address": "914 Argyle Road Alley",
              "bathrooms": 2,
              "bedrooms": 2,
              "carSpaces": 2,
              "city": "Singer",
              "index": 0,
              "latitude": -33.944576,
              "longitude": 151.25584,
              "picture": "images/tn_property01.jpg",
              "price": 937180,
            },
            Object {
              "_id": "593e9297ec4cca9c56bf61af",
              "address": "255 Raleigh Place",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 0,
              "city": "Machias",
              "index": 1,
              "latitude": -33.944471,
              "longitude": 151.2541,
              "picture": "images/tn_property02.jpg",
              "price": 703725,
            },
            Object {
              "_id": "593e929773c71925e5d7c11c",
              "address": "580 Amber Street",
              "bathrooms": 2,
              "bedrooms": 3,
              "carSpaces": 0,
              "city": "Bend",
              "index": 2,
              "latitude": -33.944644,
              "longitude": 151.24834,
              "picture": "images/tn_property03.jpg",
              "price": 837111,
            },
            Object {
              "_id": "593e92973d4d15eedb129aee",
              "address": "854 Varanda Place",
              "bathrooms": 2,
              "bedrooms": 3,
              "carSpaces": 0,
              "city": "Ivanhoe",
              "index": 3,
              "latitude": -33.940396,
              "longitude": 151.2469,
              "picture": "images/tn_property04.jpg",
              "price": 648223,
            },
            Object {
              "_id": "593e929728b7f8543b7907e1",
              "address": "857 Rockaway Parkway",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 1,
              "city": "Lydia",
              "index": 4,
              "latitude": -33.944562,
              "longitude": 151.2503,
              "picture": "images/tn_property05.jpg",
              "price": 771826,
            },
            Object {
              "_id": "593e92975996a6263d146444",
              "address": "543 Harman Street",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 1,
              "city": "Rosedale",
              "index": 5,
              "latitude": -33.939304,
              "longitude": 151.25046,
              "picture": "images/tn_property01.jpg",
              "price": 686238,
            },
            Object {
              "_id": "593e9297a13e246da5ba0884",
              "address": "787 Gerald Court",
              "bathrooms": 2,
              "bedrooms": 2,
              "carSpaces": 2,
              "city": "Allison",
              "index": 6,
              "latitude": -33.937819,
              "longitude": 151.25774,
              "picture": "images/tn_property02.jpg",
              "price": 595377,
            },
            Object {
              "_id": "593e9297afa51a013fcd07d5",
              "address": "416 Vandervoort Avenue",
              "bathrooms": 1,
              "bedrooms": 4,
              "carSpaces": 1,
              "city": "Nicut",
              "index": 7,
              "latitude": -33.943154,
              "longitude": 151.24687,
              "picture": "images/tn_property03.jpg",
              "price": 696189,
            },
            Object {
              "_id": "593e92972bd8fd05f2ab1a89",
              "address": "437 Mill Street",
              "bathrooms": 1,
              "bedrooms": 3,
              "carSpaces": 0,
              "city": "Canterwood",
              "index": 8,
              "latitude": -33.943723,
              "longitude": 151.25278,
              "picture": "images/tn_property04.jpg",
              "price": 716114,
            },
            Object {
              "_id": "593e9297fe2e6d32c859ecf4",
              "address": "553 Friel Place",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 1,
              "city": "Cobbtown",
              "index": 9,
              "latitude": -33.942459,
              "longitude": 151.24941,
              "picture": "images/tn_property05.jpg",
              "price": 739852,
            },
            Object {
              "_id": "593e92970481ccf52f948e6d",
              "address": "687 Opal Court",
              "bathrooms": 1,
              "bedrooms": 4,
              "carSpaces": 2,
              "city": "Blue",
              "index": 10,
              "latitude": -33.939606,
              "longitude": 151.25289,
              "picture": "images/tn_property01.jpg",
              "price": 625917,
            },
            Object {
              "_id": "593e9297e09417746b79e1c5",
              "address": "183 Canton Court",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 0,
              "city": "Craig",
              "index": 11,
              "latitude": -33.938557,
              "longitude": 151.24773,
              "picture": "images/tn_property02.jpg",
              "price": 591236,
            },
            Object {
              "_id": "593e92978e7160fa2b9992d9",
              "address": "234 Prospect Street",
              "bathrooms": 2,
              "bedrooms": 1,
              "carSpaces": 0,
              "city": "Orovada",
              "index": 12,
              "latitude": -33.938345,
              "longitude": 151.25045,
              "picture": "images/tn_property03.jpg",
              "price": 995539,
            },
            Object {
              "_id": "593e92973459fbdfa1f1c041",
              "address": "205 Ocean Parkway",
              "bathrooms": 1,
              "bedrooms": 1,
              "carSpaces": 2,
              "city": "Clay",
              "index": 13,
              "latitude": -33.937763,
              "longitude": 151.24721,
              "picture": "images/tn_property04.jpg",
              "price": 866924,
            },
            Object {
              "_id": "593e92970306538830483e14",
              "address": "324 Degraw Street",
              "bathrooms": 2,
              "bedrooms": 2,
              "carSpaces": 2,
              "city": "Windsor",
              "index": 14,
              "latitude": -33.938662,
              "longitude": 151.2513,
              "picture": "images/tn_property05.jpg",
              "price": 810745,
            },
            Object {
              "_id": "593e92976042982766c97ca1",
              "address": "644 Ashford Street",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 0,
              "city": "Bluetown",
              "index": 15,
              "latitude": -33.940989,
              "longitude": 151.25156,
              "picture": "images/tn_property01.jpg",
              "price": 793134,
            },
            Object {
              "_id": "593e92971901e888a036a258",
              "address": "472 Wilson Street",
              "bathrooms": 1,
              "bedrooms": 1,
              "carSpaces": 1,
              "city": "Wakulla",
              "index": 16,
              "latitude": -33.943395,
              "longitude": 151.24723,
              "picture": "images/tn_property02.jpg",
              "price": 984589,
            },
            Object {
              "_id": "593e929780e6af9c11a10b90",
              "address": "455 Crown Street",
              "bathrooms": 2,
              "bedrooms": 1,
              "carSpaces": 0,
              "city": "Greer",
              "index": 17,
              "latitude": -33.941819,
              "longitude": 151.25784,
              "picture": "images/tn_property03.jpg",
              "price": 836244,
            },
            Object {
              "_id": "593e9297441033e3f818ff6b",
              "address": "459 Duffield Street",
              "bathrooms": 2,
              "bedrooms": 3,
              "carSpaces": 1,
              "city": "Wright",
              "index": 18,
              "latitude": -33.940486,
              "longitude": 151.25562,
              "picture": "images/tn_property04.jpg",
              "price": 962426,
            },
            Object {
              "_id": "593e9297964d25b59d94e8aa",
              "address": "537 Gelston Avenue",
              "bathrooms": 2,
              "bedrooms": 3,
              "carSpaces": 0,
              "city": "Manchester",
              "index": 19,
              "latitude": -33.942826,
              "longitude": 151.24943,
              "picture": "images/tn_property05.jpg",
              "price": 768916,
            },
            Object {
              "_id": "593e929737636c879539416d",
              "address": "689 Orange Street",
              "bathrooms": 1,
              "bedrooms": 1,
              "carSpaces": 1,
              "city": "Ballico",
              "index": 20,
              "latitude": -33.943478,
              "longitude": 151.24779,
              "picture": "images/tn_property01.jpg",
              "price": 956974,
            },
            Object {
              "_id": "593e9297b2516c3e5af4601f",
              "address": "874 Rugby Road",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 2,
              "city": "Tuskahoma",
              "index": 21,
              "latitude": -33.93817,
              "longitude": 151.25545,
              "picture": "images/tn_property02.jpg",
              "price": 875917,
            },
            Object {
              "_id": "593e92972f6034590df04c45",
              "address": "687 Alabama Avenue",
              "bathrooms": 1,
              "bedrooms": 3,
              "carSpaces": 2,
              "city": "Temperanceville",
              "index": 22,
              "latitude": -33.938195,
              "longitude": 151.25286,
              "picture": "images/tn_property03.jpg",
              "price": 913858,
            },
            Object {
              "_id": "593e9297aa0e233a421f1a58",
              "address": "739 Vermont Street",
              "bathrooms": 2,
              "bedrooms": 2,
              "carSpaces": 0,
              "city": "Cliffside",
              "index": 23,
              "latitude": -33.94209,
              "longitude": 151.25645,
              "picture": "images/tn_property04.jpg",
              "price": 622995,
            },
            Object {
              "_id": "593e92977526a30b580dcfb2",
              "address": "735 Humboldt Street",
              "bathrooms": 2,
              "bedrooms": 2,
              "carSpaces": 1,
              "city": "Sunbury",
              "index": 24,
              "latitude": -33.944276,
              "longitude": 151.25442,
              "picture": "images/tn_property05.jpg",
              "price": 593777,
            },
            Object {
              "_id": "593e92975141c74b24375bd9",
              "address": "617 Charles Place",
              "bathrooms": 1,
              "bedrooms": 3,
              "carSpaces": 1,
              "city": "Gracey",
              "index": 25,
              "latitude": -33.939279,
              "longitude": 151.25559,
              "picture": "images/tn_property01.jpg",
              "price": 757686,
            },
            Object {
              "_id": "593e9297b3ffeb95bb008453",
              "address": "786 George Street",
              "bathrooms": 1,
              "bedrooms": 3,
              "carSpaces": 0,
              "city": "Farmers",
              "index": 26,
              "latitude": -33.943311,
              "longitude": 151.25781,
              "picture": "images/tn_property02.jpg",
              "price": 976878,
            },
            Object {
              "_id": "593e9297e262c91acd91ab81",
              "address": "711 Sunnyside Avenue",
              "bathrooms": 2,
              "bedrooms": 1,
              "carSpaces": 0,
              "city": "Mapletown",
              "index": 27,
              "latitude": -33.943475,
              "longitude": 151.24929,
              "picture": "images/tn_property03.jpg",
              "price": 804859,
            },
            Object {
              "_id": "593e9297df21e62e0465f91f",
              "address": "403 Lawrence Avenue",
              "bathrooms": 1,
              "bedrooms": 2,
              "carSpaces": 2,
              "city": "Snyderville",
              "index": 28,
              "latitude": -33.943007,
              "longitude": 151.25262,
              "picture": "images/tn_property04.jpg",
              "price": 611018,
            },
            Object {
              "_id": "593e929787f567dccda86910",
              "address": "265 Ocean Court",
              "bathrooms": 1,
              "bedrooms": 1,
              "carSpaces": 1,
              "city": "Inkerman",
              "index": 29,
              "latitude": -33.943142,
              "longitude": 151.25074,
              "picture": "images/tn_property05.jpg",
              "price": 707939,
            },
          ]
        }
        paginateBy={7}
        startFrom={0}
      >
        <List
          list={Array []}
        />
      </PaginatedList>
      <Pagination
        eventHandler={[Function]}
        howMany={5}
      />
    </div>
  }
/>
`;

exports[`should set init value and update setState 1`] = `
<div>
  <button
    className="ball"
    key="ball0"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="ball"
    key="ball1"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
</div>
`;
